odl/solvers/advanced/proximal_operators.py

```diff
@@ -369,15 +730,57 @@ def _call(self, x, out):
                 out.ufunc.maximum(lam, out=out)
 
                 # Global scaling
-                out /= lam
+                out /= (lam)
 
                 # Pointwise division
                 out.divide(diff, out)
 
     return _ProximalConvConjL1
 
 
-# TODO: move notes to ODL doc
+def proximal_l1(space, lam=1, g=None):
+    """Proximal operator factory of the l1-semi-norm.
+
+    Function for the proximal operator of the functional F where F is an
+    l1-semi-norm
+
+        F(x) = lam || ||x-g||_p ||_1
+
+    with x and g elements in ``space``, scaling factor lam, and point-wise
+    magnitude ||x||_p of x. If x is vector-valued, ||x||_p is the point-wise
```

Comment:
Because this code implicitly does reductions for productspaces, This should be changed to be chosable, perhaps with its own parameter (instead of doing this implicitly depending on if `space` is a `ProductSpace`.

Suggestions for name of parameter:

* reduce
* inner_norm
* more?

I wont touch this in the meanwhile.